library/private/options_tools.h

```diff
@@ -490,6 +490,28 @@ colormap_t parse(const std::string& str)
   return colormap_t(colormapVec);
 }
 
+//----------------------------------------------------------------------------
+/**
+ *  Parse provided string into a transform2d_t
+ *  Supported format: double, double, double, ... as a sequence of 9 values
+ */
+template<>
+transform2d_t parse(const std::string& str)
+{
+  std::vector<double> input;
+
+  try
+  {
+    input = parse<std::vector<double>>(str);
+  }
+  catch (const options::parsing_exception&)
+  {
+    // Quiet catch
```

Comment:
I dont think this is correct. You should either not catch it or rethrow with a more descriptive error message.

You can easily test this in TestSDKOptionsIO by using the outOfRangeDoubleStr